lib/fixed-points.nix

```diff
@@ -24,6 +24,16 @@ rec {
   # for a concrete example.
   fix' = f: let x = f x // { __unfix__ = f; }; in x;
 
+  # Return the fixpoint that `f` converges to when called recursively, starting
+  # with the input `x`.
+  #
+  #     nix-repl> converge (x: x / 2) 16
+  #     0
+  converge = f: x:
+    if (f x) == x
```

Comment:
@alyssais 
This could be optimized by introducing a let binding, i.e.

```
converge = f: x:
  let f_of_x = f x;
  in if f_of_x == x
    then x
    else converge f f_of_x;
```

That way `f x` doesn't have to be evaluated twice.